java/vector/src/test/java/org/apache/arrow/vector/ValueVectorDataPopulator.java

```diff
@@ -0,0 +1,379 @@
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TimeStampMicroTZVector vector, Long... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TimeStampMicroVector vector, Long... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TimeStampMilliTZVector vector, Long... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TimeStampMilliVector vector, Long... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TimeStampNanoTZVector vector, Long... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TimeStampNanoVector vector, Long... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TimeStampSecTZVector vector, Long... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TimeStampSecVector vector, Long... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(TinyIntVector vector, Byte... values) {
+    final int length = values.length;
+    vector.allocateNew(length);
+    vector.setValueCount(length);
+    for (int i = 0; i < length; i++) {
+      if (values[i] != null) {
+        vector.set(i, values[i]);
+      }
+    }
+  }
+
+  public static void setVector(UInt1Vector vector, Byte... values) {
```

Comment:
this is fine for now, in the future we might want to add support for the next size up to support the full range of values easily (same goes for Uint4 and Uint8